sdk/servicebus/service-bus/src/receivers/receiverCommon.ts

```diff
@@ -155,28 +168,47 @@ export function deadLetterMessage(
   const dispositionStatusOptions: DispositionStatusOptions = {
     propertiesToModify: actualPropertiesToModify,
     deadLetterReason: propertiesToModify?.deadLetterReason,
-    deadLetterDescription: propertiesToModify?.deadLetterErrorDescription
+    deadLetterDescription: propertiesToModify?.deadLetterErrorDescription,
+    retryOptions
   };
 
-  return settleMessage(
-    message,
-    DispositionType.deadletter,
-    context,
-    entityPath,
-    dispositionStatusOptions
-  );
+  return settleMessage(message, DispositionType.deadletter, context, entityPath, {
+    ...dispositionStatusOptions,
+    retryOptions
+  });
 }
 
 /**
  * @internal
- *
  */
-function settleMessage(
+export function settleMessage(
   message: ServiceBusMessageImpl,
   operation: DispositionType,
   context: ConnectionContext,
   entityPath: string,
-  options?: DispositionStatusOptions
+  options: DispositionStatusOptions,
+  _settleMessageOperation: typeof settleMessageOperation = settleMessageOperation
+): Promise<void> {
+  return retry({
+    connectionId: context.connectionId,
+    operation: () => {
+      return _settleMessageOperation(message, operation, context, entityPath, options);
+    },
+    operationType: RetryOperationType.receiverLink,
```

Comment:
We should add a new RetryOperationType for settling messages in core-amqp and use that here.

Reply:
I can add it. 

Looking at it more though I think we should consider allowing us to inject a logger, which would be more useful.

Reply:
Time to log an issue! Honestly I wish this was just a string because it only gets used for logging purposes.

Are we adding the new type in this PR or a separate one?

Reply:
Operation type added.